R/fill.R

```diff
@@ -82,22 +83,39 @@
 #'   3, "Danielle",    "Observer",   NA
 #' )
 #'
-#' # The values are inconsistently missing by position within the group
-#' # Use .direction = "downup" to fill missing values in both directions
+#' # The values are inconsistently missing by position within the `group`.
+#' # Use `.direction = "downup"` to fill missing values in both directions
+#' # and `.by = group` to apply the fill per group.
+#' squirrels %>%
+#'   fill(n_squirrels, .direction = "downup", .by = group)
+#'
+#' # If you want, you can also supply a data frame grouped with `group_by()`,
+#' # but don't forget to `ungroup()`!
 #' squirrels %>%
 #'   dplyr::group_by(group) %>%
 #'   fill(n_squirrels, .direction = "downup") %>%
 #'   dplyr::ungroup()
-#'
-#' # Using `.direction = "updown"` accomplishes the same goal in this example
-fill <- function(data, ..., .direction = c("down", "up", "downup", "updown")) {
-  check_dots_unnamed()
+fill <- function(data,
+                 ...,
+                 .direction = c("down", "up", "downup", "updown")) {
   UseMethod("fill")
 }
 
+#' @inheritParams dplyr::mutate
+#' @rdname fill
 #' @export
-fill.data.frame <- function(data, ..., .direction = c("down", "up", "downup", "updown")) {
-  vars <- tidyselect::eval_select(expr(c(...)), data, allow_rename = FALSE)
+fill.data.frame <- function(data,
+                            ...,
+                            .direction = c("down", "up", "downup", "updown"),
+                            .by = NULL) {
+  # Must be here rather than in the generic due to the placement of `.by`
+  check_dots_unnamed()
```

Comment:
Had to move this here out of the generic. Since `.by` isn't in the generic, supplying it will cause `check_dots_unnamed()` to yell at you if its in the generic.

This means that S3 method authors should add `check_dots_unnamed()` into their methods now, if they want the same strictness.